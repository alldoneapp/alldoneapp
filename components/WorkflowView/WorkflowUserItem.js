import React from 'react'
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import Icon from '../Icon'
import styles, { colors, hexColorToRGBa } from '../styles/global'

export default function WorkflowUserItem({ user, selected, closePopover, active }) {
    const select = () => {
        const step = {
            uid: user.uid,
            photoURL: user.photoURL,
            displayName: user.displayName,
        }
        closePopover(step)
    }

    return (
        <TouchableOpacity style={[localStyles.container, active && localStyles.activeContainer]} onPress={select}>
            <View style={localStyles.userContainer}>
                <Image
                    source={{ uri: user.photoURL }}
                    style={[localStyles.userImage, active && localStyles.userImageSelected]}
                />
                <Text style={[styles.subtitle1, localStyles.userName, active && localStyles.userNameSelected]}>
                    {user.displayName}
                </Text>
            </View>
            {selected && (
                <View style={[localStyles.checkContainer]}>
                    <Icon name="check" size={24} color="white" />
                </View>
            )}
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        padding: 8,
        alignItems: 'center',
    },
    activeContainer: {
        backgroundColor: hexColorToRGBa(colors.Text03, 0.16),
        borderRadius: 4,
        marginHorizontal: 0,
        paddingHorizontal: 8,
    },
    userContainer: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    userImage: {
        width: 32,
        height: 32,
        marginRight: 8,
        borderRadius: 100,
    },
    userImageSelected: {
        borderWidth: 2,
        borderColor: colors.Primary100,
    },
    checkContainer: {
        marginLeft: 'auto',
    },
    userName: {
        color: 'white',
    },
    userNameSelected: {
        color: colors.Primary100,
    },
})
