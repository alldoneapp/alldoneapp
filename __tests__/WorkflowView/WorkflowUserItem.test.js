import React from 'react'
import WorkflowUserItem from '../../components/WorkflowView/WorkflowUserItem'

import renderer from 'react-test-renderer'

jest.mock('../../utils/BackendBridge')
jest.mock('firebase', () => ({ firestore: {} }))

describe('WorkflowUserItem component', () => {
    const user = { uid: '0', displayName: 'asd', photoURL: 'a' }
    describe('WorkflowUserItem snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<WorkflowUserItem user={user} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
