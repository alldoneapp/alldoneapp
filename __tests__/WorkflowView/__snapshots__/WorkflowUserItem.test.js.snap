// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowUserItem component WorkflowUserItem snapshot test should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "a",
        }
      }
      style={
        Object {
          "borderRadius": 100,
          "height": 32,
          "marginRight": 8,
          "width": 32,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto-Medium",
            "fontSize": 16,
            "letterSpacing": 0.16,
            "lineHeight": 24,
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      asd
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": "auto",
        "opacity": 0,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "alldone",
            "fontSize": 24,
            "opacity": 1,
          }
        }
      >
        юдн
      </Text>
    </View>
  </View>
</View>
`;
